server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -385,6 +388,10 @@ public void trimUnreferencedReaders() throws IOException {
             return;
         }
 
+        if (relocatingSupplier.getAsBoolean() == true) {
+            return;
+        }
+
```

Comment:
Lets move this before the below code -
```
if (remoteGenerationDeletionPermits.tryAcquire(REMOTE_DELETION_PERMITS) == false) {
            return;
        }
```

Reply:
good catch. This permits has made the code prone to errors.